packages/gestalt/src/ModuleTitle.js

```diff
@@ -1,62 +1,56 @@
 // @flow strict
-import { type Node } from 'react';
+import type { Node } from 'react';
 import Badge from './Badge.js';
 import Box from './Box.js';
 import Flex from './Flex.js';
 import Icon from './Icon.js';
 import Text from './Text.js';
-import { type BaseModuleTitleProps, type TypeOptions } from './moduleTypes.js';
-
-type Props = {|
-  ...BaseModuleTitleProps,
-  title: string, // overwriting to be required
-  type: TypeOptions, // overwriting to be required
-|};
+import type { ModuleTitleProps } from './moduleTypes.js';
 
 /**
  * https://gestalt.pinterest.systems/Module
  */
-export default function ModuleTitle({
-  badgeText,
-  icon,
-  iconAccessibilityLabel,
-  title,
-  type = 'info',
-}: Props): Node {
-  const TYPE_ICON_ATTRIBUTES = {
-    info: {
-      icon,
-      color: 'darkGray',
-    },
-    error: {
-      icon: 'workflow-status-problem',
-      color: 'red',
-    },
-  };
+export default function ModuleTitle(props: ModuleTitleProps): Node {
+  const { iconAccessibilityLabel = '', title, type = 'info' } = props;
 
-  const { color, icon: iconName } = TYPE_ICON_ATTRIBUTES[type];
+  const decoration = ['icon', 'badgeText', 'iconButton'].find((prop) => !!props[prop]);
```

Comment:
Oh, that's really clever — I don't think I ever would have thought of this approach. But I still find it very readable. Nice!